sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/DefaultOfferRequirementProvider.java

```diff
@@ -570,7 +570,37 @@ private static boolean isPortResource(Protos.Resource resource) {
     }
 
     private static Protos.NetworkInfo getNetworkInfo(NetworkSpec networkSpec) {
-        return Protos.NetworkInfo.newBuilder().setName(networkSpec.getName()).build();
+        LOGGER.info("Loading NetworkInfo for network named \"{}\"", networkSpec.getName());
+        Protos.NetworkInfo.Builder netInfoBuilder = Protos.NetworkInfo.newBuilder();
+        netInfoBuilder.setName(networkSpec.getName());
+
+        if (!networkSpec.getPortMappings().isEmpty()) {
+            for (Map.Entry<Integer, Integer> e : networkSpec.getPortMappings().entrySet()) {
+                Integer hostPort = e.getKey();
+                Integer containerPort = e.getValue();
+                netInfoBuilder.addPortMappings(Protos.NetworkInfo.PortMapping.newBuilder()
+                        .setHostPort(hostPort)
+                        .setContainerPort(containerPort)
+                        .build());
+            }
+        }
+
+        if (!networkSpec.getNetgroups().isEmpty()) {
+            netInfoBuilder.addAllGroups(networkSpec.getNetgroups());
+        }
+
+        if (!networkSpec.getIpAddresses().isEmpty()) {
+            Collection<Protos.NetworkInfo.IPAddress> ipAddresses = new ArrayList<>();
+            for (String ipAddressString : networkSpec.getIpAddresses()) {
+                ipAddresses.add(Protos.NetworkInfo.IPAddress.newBuilder()
```

Comment:
This is mostly a matter of preference, but you can also create a new builder of some sub-object kept in a list on a protobuf directly on the top-level builder itself, in this case something like `netInfoBuilder.addIpAddressBuilder().etc()`. It would save the need to maintain an intermediate collection of IP addresses.